src/debug.c

```diff
@@ -1675,23 +1675,15 @@ void logStackTrace(void *eip, int uplevel) {
 
 /* Log global server info */
 void logServerInfo(void) {
-    
-    if (server.current_client == NULL) return;
-    
-    client *cc = server.current_client;
     sds infostring, clients;
     serverLogRaw(LL_WARNING|LL_RAW, "\n------ INFO OUTPUT ------\n");
-    int all_sections = 0;
-    int everything = 0;
-    dict *sections_dict = genInfoSectionDict(cc, "server", &all_sections, &everything);
-    infostring = genRedisInfoString(sections_dict,1,0);
+    infostring = genRedisInfoString(NULL,1,0);
```

Comment:
didn't we say that passing some empty dict is better here?
i.e. the code is not a time critical code, so there's no down side of another allocation or extra call to `genInfoSectionDict`.
on the other hand if this leads to some code path that's not normally executed (not covered by the test suite), who doesn't have a check for a null dict (e.g maybe module.c) it'll blow up

Reply:
Hi Oran, Yes, we agreed with here to use an empty dict. But when I do coding practice, I found creating an empty dict makes thing complex. 
Because the second parameter is 1 which means "all", then in our code genRedisInfoString function (server.c), it has the short circuit statement:  **if (all_sections || (dictFind(section_dict,"server") != NULL))**
. It means even we only pass a NULL as parameter in the **genRedisInfoString(NULL,1,0)**; it does not have harmful to this function.  
This is why I change mind to pass NULL instead of passing an empty dict, Do you agree with this idea? Thanks.

Reply:
i'm afraid we'll miss some check in some `if` and that this code will not be covered by the test suite.
since this is a crash report code, i don't think performance matters, and i'll feel better passing an empty dict.